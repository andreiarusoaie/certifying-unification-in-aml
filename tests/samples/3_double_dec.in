variables: y
symbols: f, g, a
problem: f(g(a)) =? f(g(y))
