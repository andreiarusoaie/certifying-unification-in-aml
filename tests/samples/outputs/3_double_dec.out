Found Maude version: 3.0beta1

[1m[95mProof of:[94m (f(g(a)) or f(g(y)) <--> âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y)))  [0m
(1)(âˆƒ v(5) . (f(g(a)) <--> v(5) and (v(5) === f(g(a))))) [e-subst] ;

(2)((f(g(a)) <--> âˆƒ v(5) . v(5) and (v(5) === f(g(a)))) <--> âˆƒ v(5) . (f(g(a)) <--> v(5) and (v(5) === f(g(a))))) [e-scope] ;

(3)(f(g(a)) <--> âˆƒ v(5) . v(5) and (v(5) === f(g(a)))) [modus-ponens(1, 2)] ;

(4)(âˆƒ v(5) . (f(g(y)) <--> v(5) and (v(5) === f(g(y))))) [e-subst] ;

(5)((f(g(y)) <--> âˆƒ v(5) . v(5) and (v(5) === f(g(y)))) <--> âˆƒ v(5) . (f(g(y)) <--> v(5) and (v(5) === f(g(y))))) [e-scope] ;

(6)(f(g(y)) <--> âˆƒ v(5) . v(5) and (v(5) === f(g(y)))) [modus-ponens(4, 5)] ;

(7)(f(g(a)) or f(g(y)) <--> (âˆƒ v(5) . v(5) and (v(5) === f(g(a)))) or (âˆƒ v(5) . v(5) and (v(5) === f(g(y))))) [eqv-or(3, 6)] ;

(8)((âˆƒ v(5) . v(5) and (v(5) === f(g(a)))) or (âˆƒ v(5) . v(5) and (v(5) === f(g(y)))) <--> âˆƒ v(5) . v(5) and (v(5) === f(g(a))) or v(5) and (v(5) === f(g(y)))) [e-collapse] ;

(9)(v(5) and ((v(5) === f(g(a))) or (v(5) === f(g(y)))) <--> v(5) and (v(5) === f(g(a))) or v(5) and (v(5) === f(g(y)))) [and-or-distr] ;

(10)((âˆƒ v(5) . v(5) and ((v(5) === f(g(a))) or (v(5) === f(g(y))))) <--> âˆƒ v(5) . v(5) and (v(5) === f(g(a))) or v(5) and (v(5) === f(g(y)))) [e-intro(9)] ;

(11)(f(g(a)) or f(g(y)) <--> âˆƒ v(5) . v(5) and (v(5) === f(g(a))) or v(5) and (v(5) === f(g(y)))) [eqv-tranz(7, 8)] ;

(12)(f(g(a)) or f(g(y)) <--> âˆƒ v(5) . v(5) and ((v(5) === f(g(a))) or (v(5) === f(g(y))))) [eqv-tranz(11, 10)] ;

(13)((v(5) === f(g(a))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a))) [e-gen] ;

(14)((v(5) === f(g(a))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a))) [and-eqv-intro(13)] ;

(15)((âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a))) [e-scope] ;

(16)((v(5) === f(g(a))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a))) [eqv-tranz(14, 15)] ;

(17)((v(5) === f(g(y))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(y))) [e-gen] ;

(18)((v(5) === f(g(y))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(y))) [and-eqv-intro(17)] ;

(19)((âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(y))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(y))) [e-scope] ;

(20)((v(5) === f(g(y))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(y))) [eqv-tranz(18, 19)] ;

(21)((v(5) === f(g(a))) or (v(5) === f(g(y))) <--> (âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a))) or (âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(y)))) [eqv-or(16, 20)] ;

(22)((âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a))) or (âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(y))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a)) or (v(5) === f(v(8))) and (v(8) === g(y))) [e-collapse] ;

(23)((v(5) === f(g(a))) or (v(5) === f(g(y))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a)) or (v(5) === f(v(8))) and (v(8) === g(y))) [eqv-tranz(21, 22)] ;

(24)(((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8))) <--> (v(5) === f(v(8))) and (v(8) === g(a)) or (v(5) === f(v(8))) and (v(8) === g(y))) [and-or-distr] ;

(25)((âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) <--> âˆƒ v(8) . (v(5) === f(v(8))) and (v(8) === g(a)) or (v(5) === f(v(8))) and (v(8) === g(y))) [e-intro(24)] ;

(26)((v(5) === f(g(a))) or (v(5) === f(g(y))) <--> âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [eqv-tranz(23, 25)] ;

(27)((âˆƒ v(5) . v(5) and ((v(5) === f(g(a))) or (v(5) === f(g(y))))) <--> âˆƒ v(5) . v(5) and (âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8))))) [e-context(26)] ;

(28)(v(5) and (âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) <--> âˆƒ v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [e-scope] ;

(29)((âˆƒ v(5) . v(5) and (âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8))))) <--> âˆƒ v(5) . âˆƒ v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [e-intro(28)] ;

(30)((âˆƒ v(5) . âˆƒ v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) <--> âˆƒ v(5) ; v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [e-set] ;

(31)((âˆƒ v(5) . v(5) and (âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8))))) <--> âˆƒ v(5) ; v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [eqv-tranz(29, 30)] ;

(32)(((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8)))) <--> âˆƒ v(5) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [e-scope] ;

(33)((âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) <--> âˆƒ v(8) . âˆƒ v(5) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [e-intro(32)] ;

(34)((âˆƒ v(8) . âˆƒ v(5) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) <--> âˆƒ v(5) ; v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [e-set] ;

(35)((âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) <--> âˆƒ v(5) ; v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [eqv-tranz(34, 33)] ;

(36)(âˆƒ v(5) . (f(v(8)) <--> v(5) and (v(5) === f(v(8))))) [e-subst] ;

(37)((f(v(8)) <--> âˆƒ v(5) . v(5) and (v(5) === f(v(8)))) <--> âˆƒ v(5) . (f(v(8)) <--> v(5) and (v(5) === f(v(8))))) [e-scope] ;

(38)(f(v(8)) <--> âˆƒ v(5) . v(5) and (v(5) === f(v(8)))) [modus-ponens(36, 37)] ;

(39)((âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) [e-context(38)] ;

(40)((âˆƒ .Vars . âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ .Vars . âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) [e-intro(39)] ;

(41)((âˆƒ .Vars . âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) <--> âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) [e-set] ;

(42)((âˆƒ .Vars . âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) [e-set] ;

(43)((âˆƒ .Vars . âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) [eqv-tranz(41, 40)] ;

(44)((âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) [eqv-tranz(43, 42)] ;

(45)((âˆƒ v(5) . v(5) and ((v(5) === f(g(a))) or (v(5) === f(g(y))))) <--> âˆƒ v(5) ; v(8) . v(5) and ((v(8) === g(a)) or (v(8) === g(y))) and (v(5) === f(v(8)))) [eqv-tranz(27, 31)] ;

(46)((âˆƒ v(5) . v(5) and ((v(5) === f(g(a))) or (v(5) === f(g(y))))) <--> âˆƒ v(8) . ((v(8) === g(a)) or (v(8) === g(y))) and (âˆƒ v(5) . v(5) and (v(5) === f(v(8))))) [eqv-tranz(45, 35)] ;

(47)((âˆƒ v(5) . v(5) and ((v(5) === f(g(a))) or (v(5) === f(g(y))))) <--> âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) [eqv-tranz(44, 46)] ;

(48)(f(g(a)) or f(g(y)) <--> âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) [eqv-tranz(12, 47)] ;

(49)((v(8) === g(a)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a)) [e-gen] ;

(50)((v(8) === g(a)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a)) [and-eqv-intro(49)] ;

(51)((âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a)) [e-scope] ;

(52)((v(8) === g(a)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a)) [eqv-tranz(50, 51)] ;

(53)((v(8) === g(y)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === y)) [e-gen] ;

(54)((v(8) === g(y)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === y)) [and-eqv-intro(53)] ;

(55)((âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === y)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === y)) [e-scope] ;

(56)((v(8) === g(y)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === y)) [eqv-tranz(54, 55)] ;

(57)((v(8) === g(a)) or (v(8) === g(y)) <--> (âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a)) or (âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === y))) [eqv-or(52, 56)] ;

(58)((âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a)) or (âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === y)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a) or (v(8) === g(v(9))) and (v(9) === y)) [e-collapse] ;

(59)((v(8) === g(a)) or (v(8) === g(y)) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a) or (v(8) === g(v(9))) and (v(9) === y)) [eqv-tranz(57, 58)] ;

(60)(((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9))) <--> (v(8) === g(v(9))) and (v(9) === a) or (v(8) === g(v(9))) and (v(9) === y)) [and-or-distr] ;

(61)((âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) <--> âˆƒ v(9) . (v(8) === g(v(9))) and (v(9) === a) or (v(8) === g(v(9))) and (v(9) === y)) [e-intro(60)] ;

(62)((v(8) === g(a)) or (v(8) === g(y)) <--> âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [eqv-tranz(59, 61)] ;

(63)((âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(8) . f(v(8)) and (âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9))))) [e-context(62)] ;

(64)(f(v(8)) and (âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) <--> âˆƒ v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [e-scope] ;

(65)((âˆƒ v(8) . f(v(8)) and (âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9))))) <--> âˆƒ v(8) . âˆƒ v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [e-intro(64)] ;

(66)((âˆƒ v(8) . âˆƒ v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) <--> âˆƒ v(8) ; v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [e-set] ;

(67)((âˆƒ v(8) . f(v(8)) and (âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9))))) <--> âˆƒ v(8) ; v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [eqv-tranz(65, 66)] ;

(68)(((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9)))) <--> âˆƒ v(8) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [e-scope] ;

(69)((âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) <--> âˆƒ v(9) . âˆƒ v(8) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [e-intro(68)] ;

(70)((âˆƒ v(9) . âˆƒ v(8) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) <--> âˆƒ v(8) ; v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [e-set] ;

(71)((âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) <--> âˆƒ v(8) ; v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [eqv-tranz(70, 69)] ;

(72)(âˆƒ v(8) . (f(g(v(9))) <--> f(v(8)) and (v(8) === g(v(9))))) [e-subst] ;

(73)((f(g(v(9))) <--> âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9)))) <--> âˆƒ v(8) . (f(g(v(9))) <--> f(v(8)) and (v(8) === g(v(9))))) [e-scope] ;

(74)(f(g(v(9))) <--> âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9)))) [modus-ponens(72, 73)] ;

(75)((âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) <--> âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) [e-context(74)] ;

(76)((âˆƒ .Vars . âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) <--> âˆƒ .Vars . âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) [e-intro(75)] ;

(77)((âˆƒ .Vars . âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) <--> âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) [e-set] ;

(78)((âˆƒ .Vars . âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) <--> âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) [e-set] ;

(79)((âˆƒ .Vars . âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) <--> âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) [eqv-tranz(77, 76)] ;

(80)((âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) <--> âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) [eqv-tranz(79, 78)] ;

(81)((âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(8) ; v(9) . f(v(8)) and ((v(9) === a) or (v(9) === y)) and (v(8) === g(v(9)))) [eqv-tranz(63, 67)] ;

(82)((âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(9) . ((v(9) === a) or (v(9) === y)) and (âˆƒ v(8) . f(v(8)) and (v(8) === g(v(9))))) [eqv-tranz(81, 71)] ;

(83)((âˆƒ v(8) . f(v(8)) and ((v(8) === g(a)) or (v(8) === g(y)))) <--> âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) [eqv-tranz(80, 82)] ;

(84)(f(g(a)) or f(g(y)) <--> âˆƒ v(9) . f(g(v(9))) and ((v(9) === a) or (v(9) === y))) [eqv-tranz(48, 83)] ;
[94mChecked: [0m [1m[92m true
[0m
